lib/sensors/temperature.cpp

```diff
@@ -2,81 +2,92 @@
                          std::shared_ptr<io::II2c> i2c,
-                         const std::uint8_t channel)
+                         const std::uint8_t channel,
+                         const std::uint8_t device_address)
     : logger_(logger),
       i2c_(i2c),
-      channel_(channel)
+      channel_(channel),
+      device_address_(device_address)
 {
 }
 
 Temperature::~Temperature()
 {
 }
 
-std::optional<std::int16_t> Temperature::read()
+std::optional<core::Result> Temperature::checkStatus()
 {
-  const auto status_check_result = i2c_->readByte(kTemperatureDefaultAddress, kStatus);
+  const auto status_check_result = i2c_->readByte(device_address_, kStatus);
   if (!status_check_result) {
     logger_.log(
       core::LogLevel::kFatal, "Failed to read temperature sensor status at channel %d", channel_);
     return std::nullopt;
   }
-  // TODOLater: change this from a fatal error to a debug and re-prompt read (with ROS)
-  if (status_check_result.value() == kBusy) {
-    logger_.log(core::LogLevel::kFatal,
+  if (*status_check_result == kBusy) {
+    logger_.log(core::LogLevel::kWarn,
                 "Failed to read, temperature sensor is not ready to be read at channel %d",
                 channel_);
-    return std::nullopt;
+    return core::Result::kFailure;
   }
-  if (status_check_result.value() == kTemperatureOverUpperLimit) {
+  if (*status_check_result == kTemperatureOverUpperLimit) {
     logger_.log(core::LogLevel::kFatal,
                 "Failed to read, temperature is above upper limit at channel %d",
                 channel_);
     return std::nullopt;
   }
-  if (status_check_result.value() == kTemperatureUnderLowerLimit) {
+  if (*status_check_result == kTemperatureUnderLowerLimit) {
     logger_.log(core::LogLevel::kFatal,
                 "Failed to read, temperature is below lower limit at channel %d",
                 channel_);
     return std::nullopt;
   }
-  const auto temperature_high_byte
-    = i2c_->readByte(kTemperatureDefaultAddress, kDataTemperatureHigh);
+  return core::Result::kSuccess;
+}
+
+std::optional<std::int16_t> Temperature::read()
+{
+  const auto temperature_high_byte = i2c_->readByte(device_address_, kDataTemperatureHigh);
   if (!temperature_high_byte) {
     logger_.log(
       core::LogLevel::kFatal, "Failed to read high byte for temperature at channel %d", channel_);
     return std::nullopt;
   }
-  const auto temperature_low_byte = i2c_->readByte(kTemperatureDefaultAddress, kDataTemperatureLow);
+  const auto temperature_low_byte = i2c_->readByte(device_address_, kDataTemperatureLow);
   if (!temperature_low_byte) {
     logger_.log(
       core::LogLevel::kFatal, "Failed to read low byte for temperature at channel %d", channel_);
     return std::nullopt;
   }
-  const std::int16_t temperature
-    = ((temperature_high_byte.value() << 8) | temperature_low_byte.value());
+  const std::int16_t temperature = ((*temperature_high_byte << 8) | *temperature_low_byte);
```

Comment:
not `uint16_t`?

Reply:
Given that it's just bytes at this point?

Reply:
nope, temperature can be negative (though we should never have that occur, but if we make it uint we cant check for it)